Userland/Libraries/LibGfx/ImageFormats/TIFFLoader.cpp

```diff
@@ -830,6 +837,29 @@ ErrorOr<Optional<ReadonlyBytes>> TIFFImageDecoderPlugin::icc_data()
     return m_context->metadata().icc_profile().map([](auto const& buffer) -> ReadonlyBytes { return buffer.bytes(); });
 }
 
+NaturalFrameFormat TIFFImageDecoderPlugin::natural_frame_format() const
+{
+    if (m_context->is_cmyk())
+        return NaturalFrameFormat::CMYK;
+    return NaturalFrameFormat::RGB;
+}
+
+ErrorOr<NonnullRefPtr<CMYKBitmap>> TIFFImageDecoderPlugin::cmyk_frame()
+{
+    VERIFY(natural_frame_format() == NaturalFrameFormat::CMYK);
+
+    if (m_context->state() == TIFF::TIFFLoadingContext::State::Error)
+        return Error::from_string_literal("TIFFImageDecoderPlugin: Decoding failed");
+
+    if (m_context->state() < TIFF::TIFFLoadingContext::State::FrameDecoded)
+        TRY(m_context->decode_frame());
+
+    if (m_context->state() == TIFF::TIFFLoadingContext::State::Error)
```

Comment:
we don't do this 2nd state check in line 823. should we? if yes, separate commit (or pr). if not, we probably shouldn't do that here either?

Reply:
In theory, we shouldn't as any error should bubble up from `TRY(m_context->decode_frame());`. I would be more inclined to remove it from here too. Do you remember why you added that in the first place?

Reply:
Oh, because the TRY() would return, makes sense. Removing here is probably the thing to do then.

Reply:
Ok, doing that right now.